extract.go

```diff
@@ -32,28 +32,67 @@ import (
 	"path/filepath"
 
 	"github.com/juju/errors"
+	"gopkg.in/h2non/filetype.v1"
```

Comment:
I'm not big on importing other packages. I'd like to keep the library lightweight.